pkg/controllers/mcs/service_export_controller.go

```diff
@@ -0,0 +1,493 @@
+		}
+	case common.EndpointSliceGVK:
+		if err := c.reconcileEndpointSlice(ctx, reconcileKey); err != nil {
+			keyedLogger.Error(err, fmt.Sprintf("failed to reconcile endpointSlice(%s/%s)", reconcileKey.namespace, reconcileKey.name))
+			return worker.StatusError
+		}
+	}
+
+	return worker.StatusAllOK
+}
+
+func (c *ServiceExportController) reconcileServiceExport(ctx context.Context, key reconcileKey) error {
+	logger := klog.FromContext(ctx)
+
+	needReconcile, err := c.filterServiceExport(ctx, key)
+	if err != nil {
+		return fmt.Errorf("failed to check if serviceExport needs to be filtered: %w", err)
+	}
+	if !needReconcile {
+		return nil
+	}
+
+	seLister, synced, exists := c.fedInformerManager.GetServiceExportLister(key.cluster)
+	if !exists {
+		return fmt.Errorf("informer of serviceExport not exists for cluster %s", key.cluster)
+	}
+	if !cache.WaitForCacheSync(ctx.Done(), synced) {
+		return fmt.Errorf("timeout waiting for serviceExport informer sync for cluster %s", key.cluster)
+	}
+
+	_, err = seLister.ByNamespace(key.namespace).Get(key.name)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			logger.V(2).Info("Cleanup imported endpointSlice due to serviceExport event")
+			deleteErr := c.kubeClient.DiscoveryV1beta1().EndpointSlices(key.namespace).DeleteCollection(ctx,
+				metav1.DeleteOptions{}, metav1.ListOptions{
+					LabelSelector: labels.SelectorFromSet(labels.Set{
+						discoveryv1b1.LabelServiceName:            key.name,
+						EndpointSliceSourceClusterLabel:           key.cluster,
+						managedlabel.ManagedByKubeAdmiralLabelKey: managedlabel.ManagedByKubeAdmiralLabelValue,
+					}).String(),
+				})
+			if deleteErr != nil {
+				logger.Error(deleteErr, "Failed to cleanup importedSlice in control plane")
+			}
+			return deleteErr
+		}
+		return err
+	}
+
+	clusterEpsLister, synced, exist := c.fedInformerManager.GetEndpointSliceLister(key.cluster)
+	if !exist {
+		return fmt.Errorf("informer of endpointSlice not exists for cluster %s", key.cluster)
+	}
+	if !cache.WaitForCacheSync(ctx.Done(), synced) {
+		return fmt.Errorf("timeout waiting for endpointSlice informer sync for cluster %s", key.cluster)
+	}
+
+	clusterEndpointSlices, err := clusterEpsLister.List(labels.SelectorFromSet(labels.Set{
+		discoveryv1b1.LabelServiceName: key.name,
+	}))
+	if err != nil {
+		return err
+	}
+
+	fedEpsLister := c.endpointSliceInformer.Lister()
+	var errs []error
+	for _, eps := range clusterEndpointSlices {
+		if _, exist = eps.Labels[EndpointSliceSourceClusterLabel]; exist {
+			continue
+		}
+
+		importedEps := eps.DeepCopy()
+		importedEps.ObjectMeta = metav1.ObjectMeta{
+			Namespace: eps.Namespace,
+			Name:      naming.GenerateImportedEndpointSliceName(eps.GetName(), key.cluster),
+			Labels: map[string]string{
+				discoveryv1b1.LabelServiceName:            key.name,
+				EndpointSliceSourceClusterLabel:           key.cluster,
+				managedlabel.ManagedByKubeAdmiralLabelKey: managedlabel.ManagedByKubeAdmiralLabelValue,
```

Comment:
Having a resource with managed label in the host cluster can be very confusing. Any change to use a different label and not overload the kubeadmiral managed label?